public/components/reindexController.react.js

```diff
@@ -37,10 +40,19 @@ export default class ReindexControllerComponent extends React.Component {
             this.loadContentSource(nextProps.routeParams.id, nextProps.routeParams.environment);
             this.loadReindexHistory(nextProps.routeParams.id, nextProps.routeParams.environment);
             this.loadRunningReindex(nextProps.routeParams.id, nextProps.routeParams.environment);
+            this.loadContentSourceWithId(nextProps.routeParams.id);
             this.setState({editModeOn: false});
         }
     }
 
+    loadContentSourceWithId(id) {
+        ContentSourceService.getContentSourcesWithId(id).then(response => {
+            this.setState({
+                contentSourcesForEnvironments: response.contentSources
+            });
+        });
+    }
+
     loadContentSource(id, environment) {
```

Comment:
Yes. The `loadContentSourceWithId` is used for the pills with the environments (returns a list of content sources) and the `loadContentSource` is used for actually showing the content source we want (returns only one content source that matches the route).